### PR TITLE
add HUNO tracker

### PR DESCRIPTION
**Pull request by swannie-eire**
all custom naming conventions are for this tracker is done in the huno class so wont affect others.

tested with 1 series and 2 movies and works as expected

### CODE REVIEW

File: src/trackers/HUNO.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        await self.edit_torrent(meta)
+        cat_id = await self.get_cat_id(meta['category'])
+        type_id = await self.get_type_id(meta['type'])
+        resolution_id = await self.get_res_id(meta['resolution'])
+        await self.edit_desc(meta)
+        if meta['anon'] == 0 and bool(distutils.util.strtobool(self.config['TRACKERS']['HUNO'].get('anon', "False"))) == False:
+            anon = 0
+        else:
+            anon = 1
+        if bool(distutils.util.strtobool(self.config['TRACKERS']['HUNO'].get('internal', "False"))) == False:
```
Comment (L4GSP1KE) [inline src/trackers/HUNO.py:41]: please change this internal config option to be consistent with the rest of the trackers. See below for implementation / usage: 

https://github.com/L4GSP1KE/Upload-Assistant/blob/f849e0b0e2e2ec444b42f15ec480e32122cecd76/src/trackers/AITHER.py#L83-L86 
https://github.com/L4GSP1KE/Upload-Assistant/blob/f849e0b0e2e2ec444b42f15ec480e32122cecd76/data/example-config.py#L84-L86

File: src/trackers/HUNO.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        return 0
+
+    async def edit_torrent(self, meta):
+        HUNO_torrent = Torrent.read(f"{meta['base_dir']}/tmp/{meta['uuid']}/BASE.torrent")
+        HUNO_torrent.metainfo['announce'] = self.config['TRACKERS']['HUNO']['announce_url'].strip()
+        HUNO_torrent.metainfo['info']['source'] = "HUNO"
+        Torrent.copy(HUNO_torrent).write(f"{meta['base_dir']}/tmp/{meta['uuid']}/[HUNO]{meta['clean_name']}.torrent", overwrite=True)
+        return
+
+    async def edit_desc(self, meta):
```
Comment (L4GSP1KE) [inline src/trackers/HUNO.py:253]: after changing to the shared unit3d description function you can delete this

File: src/trackers/HUNO.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        self.upload_url = 'https://hawke.uno/api/torrents/upload'
+        self.forum_link = 'https://hawke.uno/pages/1'
+        pass
+
+    async def upload(self, meta):
+        await self.edit_torrent(meta)
+        cat_id = await self.get_cat_id(meta['category'])
+        type_id = await self.get_type_id(meta['type'])
+        resolution_id = await self.get_res_id(meta['resolution'])
+        await self.edit_desc(meta)
```
Comment (L4GSP1KE) [inline src/trackers/HUNO.py:36]: the edit_desc function doesn't seem to provide anything unique, so please use the shared unit3d description function. Example:

https://github.com/L4GSP1KE/Upload-Assistant/blob/f849e0b0e2e2ec444b42f15ec480e32122cecd76/src/trackers/AITHER.py#L34

File: src/trackers/HUNO.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+                    if not any(f.lower().endswith(e) for e in video_file_extensions):
+                        continue
+                    video_files.append(f)
+
+            if len(video_files) > 1:
+                return 1
+
+        return 0
+
+    async def edit_torrent(self, meta):
```
Comment (L4GSP1KE) [inline src/trackers/HUNO.py:246]: delete after changing to the shared edit_torrent function

File: src/trackers/HUNO.py [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+        self.config = config
+        self.tracker = 'HUNO'
+        self.source_flag = 'HUNO'
+        self.search_url = 'https://hawke.uno/api/torrents/filter'
+        self.upload_url = 'https://hawke.uno/api/torrents/upload'
+        self.forum_link = 'https://hawke.uno/pages/1'
+        pass
+
+    async def upload(self, meta):
+        await self.edit_torrent(meta)
```
Comment (L4GSP1KE) [inline src/trackers/HUNO.py:32]: please use the shared edit_torrent function
https://github.com/L4GSP1KE/Upload-Assistant/blob/f849e0b0e2e2ec444b42f15ec480e32122cecd76/src/trackers/AITHER.py#L33

**Review by L4GSP1KE — CHANGES_REQUESTED** [review]
There are a couple changes I'd like to see to bring this up to date and consistent with the other unit3d sites

**Review by swannie-eire — COMMENTED** [review]
added changes

**Comment by swannie-eire** [comment]
I think i address everything you asked for.

Please have a look and let me know.

**Comment by L4GSP1KE** [comment]
Looks good. Thanks

**Pull request merged**